src/Groupper.ts

```diff
@@ -430,6 +430,7 @@ export class GroupperAPI implements Types.GroupperAPI {
     private _current: Record<string, Types.Groupper> = {};
     private _grouppers: Record<string, Types.Groupper> = {};
     private _handleKeyPressTimer: number | undefined;
+    private _asyncFocusIntent: Types.AsyncFocusIntent | undefined;
```

Comment:
Each entity only has **one** intent - there is no real reason why they need to use it to cancel and commit.

You could quite simply have `registerAsyncFocus(sourece, callback)` and `cancelAsyncFocus(source)` and have the FocusedElement state be the source of truth